### ISSUE TITLE
'shared' is unavailable in application extensions for iOS: Use view controller based solutions where appropriate instead on iOS 15 Betas.

### ISSUE THREAD

**Issue by xxTigerShark** [yeahdongcn/RSBarcodes_Swift#147]
'shared' is unavailable in application extensions for iOS: Use view controller based solutions where appropriate instead.

This is a new error popping up on the iOS 15 for SwiftUI Applications. It seems to be tied to:
`      let statusBarOrientation = UIApplication.shared.statusBarOrientation`

**Comment by yeahdongcn** [comment]
Is your project an application extension?

**Comment by xxTigerShark** [comment]
Appears I found the issue, it's a bug in Swift with Beta 3 using SPM. This line should solve it.

`@available(iOSApplicationExtension, unavailable)`

[https://forums.swift.org/t/set-application-extension-api-only-on-a-spm-package/39333/11](https://forums.swift.org/t/set-application-extension-api-only-on-a-spm-package/39333/11)

**Comment by xxTigerShark** [comment]
It sounds like this will be corrected soon, and until then you just need to use any other method to install the package so this is less of an issue than I though.

**Comment by yeahdongcn** [comment]
Let's keep this open until it is resolved officially.

**Comment by xxTigerShark** [comment]
Yep, still appears to be present in the newest betas. Curious to see if this lasts until the release.

**Comment by joshuapoq** [comment]
This is the way they are choosing to resolve it? I don't expect they'll fix this as they expect frameworks to fix this?

As stated in the patch notes:
> Linking Swift packages from application extension targets or watchOS applications no longer emits unresolvable warnings about linking to libraries not safe for use in application extensions. This means that code referencing APIs annotated as unavailable for use in app extensions must now themselves be annotated as unavailable for use in application extensions, in order to allow that code to be used in both apps and app extensions. (66928265)
